examples/0006-using-rest-api-directly/undo-playbook.yaml

```diff
@@ -0,0 +1,93 @@
+---
+
+- name: Use the REST API directly
+  hosts: f5-test
+  connection: local
+
+  tasks:
+    - name: Get an API token
+      uri:
+        url: "https://{{ ansible_host }}:{{ bigip_port }}/mgmt/shared/authn/login"
+        method: POST
+        body: |
+          {
+            "username": "{{ bigip_username }}",
+            "password": "{{ bigip_password }}",
+            "loginProviderName": "tmos"
+          }
+        force_basic_auth: yes
+        status_code: 200
+        body_format: json
+        validate_certs: "{{ validate_certs }}"
+      register: auth
+
+    - name: Check for a virtual server
+      uri:
+        url: "https://{{ ansible_host }}:{{ bigip_port }}/mgmt/tm/ltm/virtual/~Common~foo"
+        method: GET
+        headers:
+          X-F5-Auth-Token: "{{ auth['json']['token']['name'] }}"
+        status_code: 200, 404
+        validate_certs: "{{ validate_certs }}"
+      register: virtual
+
+    - debug:
+        msg: "{{virtual.status}}"
+
+    - name: Delete a virtual server
+      uri:
+        url: "https://{{ ansible_host }}:{{ bigip_port }}/mgmt/tm/ltm/virtual/~Common~foo"
+        method: DELETE
+        headers:
+          X-F5-Auth-Token: "{{ auth['json']['token']['name'] }}"
+        status_code: 200
+        validate_certs: "{{ validate_certs }}"
+      when: virtual.status == "404"
+
+    ###  when you delete a pool the pool members will be deleted
```

Comment:
I would remove this commented code and instead put a comment at the top of the playbook that tells the user what the playbook does, and mentions that deleting the pool member is not necessary